src/Events/Category_Colors/Controller.php

```diff
@@ -23,12 +30,30 @@
  */
 class Controller extends Controller_Contract {
 
+	use Migration_Trait;
+
 	/**
 	 * Register the provider.
 	 *
 	 * @since TBD
 	 */
 	public function do_register(): void {
+		$this->container->singleton( Pre_Processor::class );
+		$this->container->singleton( Validator::class );
+		$this->container->singleton( Migration_Runner::class );
+		$this->container->singleton( Post_Processor::class );
+		$this->container->singleton( Logger::class );
+		$this->container->singleton( Migration_Process::class );
+		$this->add_filters();
+	}
+
+	/**
+	 * Adds the filters required.
+	 *
+	 * @since TBD
+	 */
+	protected function add_filters() {
+		add_action( 'admin_init', [ $this, 'debug_migration_process' ] );
```

Comment:
remove it in unregister as well.

If you had a controller_test_case for this controller there is an automatic test that catches that you are registering a different amount of hooks that you are unregistering.